easysnmp/session.py

```diff
@@ -203,14 +204,20 @@ def __init__(
         abort_on_nonexistent=False,
     ):
         # Validate and extract the remote port
-        if ":" in hostname:
+        connection_string = re.match(
+            r"^((?:tcp6?:|udp6?:)?\[?([^\[\]]+?)\]?)\:(\d+)$", hostname
+        )
+        if connection_string:
             if remote_port:
                 raise ValueError(
                     "a remote port was specified yet the hostname appears "
                     "to have a port defined too"
                 )
             else:
-                hostname, remote_port = hostname.split(":")
+                full_address, hostname, remote_port = connection_string.groups()
+                if ":" in hostname and not hostname_is_IPv6(hostname):
+                    raise ValueError("an invalid IPv6 address was specified")
```

Comment:
Hi, this exception is also thrown for valid IPv6 addresses such as "2001:8000:0103:0982:0000:0000:0000:0002". The problem is that it ends in a digit, which is interpreted as the remote_port by line 217 and the regex on line 208. 

It will work if the IPv6 address was "2001:8000:0103:0982:0000:0000:0000:abcd" for example, because `abcd` will never be seen as a digit by the regex.

Reply:
You are right! I'll update the testcase and the code accordingly as soon as i can.